public/js/index.js

```diff
@@ -524,6 +524,62 @@ Index.loadContextMenuCategories = (catList, id) => Server.callAPI(`/api/archives
     },
 );
 
+/**
+ * Build rating options for contextMenu and select the one for the current ID.
+ * @param {*} id The ID of the archive to check
+ * @returns Ratings
+ */
+Index.loadContextMenuRatings = (id) => Server.callAPI(`/api/archives/${id}/metadata`, "GET", null, `Error finding metadata for ${id}!`,
+    (data) => {
+        const items = {};
+        const ratings = [{
+            id: 1,
+            name: "⭐",
+        }, {
+            id: 2,
+            name: "⭐⭐",
+        }, {
+            id: 3,
+            name: "⭐⭐⭐",
+        }, {
+            id: 4,
+            name: "⭐⭐⭐⭐",
+        }, {
+            id: 5,
+            name: "⭐⭐⭐⭐⭐",
+        }];
+        const tags = data.tags.split(",").map((x) => x.split(":"));
```

Comment:
You can use `LRR.splitTagsByNamespace(data.tags)` here, then directly look if the "rating" key exists to get the existing rating tag. I think it should also make the underlying logic easier to write.